### PR TITLE
Beamline attributes, movables and actuators

### PR DESCRIPTION
**Pull request by marcus-oscarsson**
Hello !

This is various fixes related to the actuators and movables. They were all working a while back a go when tested on a beamline, various refactoring and updates have then broken the functionality slightly. I could basically not use the controls to control the "movables" (energy, resolution ...). For those I could set the intermediate values (no value changes) were not handled.

So I have merged actuators and movables into one concept called attributes (as in beamline attributes), the name is maybe poorly chosen. It can definitely be changed if someone have a better suggestion. The reason for doing this was simply that the actuators and movables where all handled the same by the backend logic so there is no need to add extra logic to separate them in the UI or in the backend for that matter.

Ive added the machine info in the same way (as an attribute), and added the necessary logic to handle it.
The cryo attribute used a custom made CryoInput which I changed to a PopInput, I dont know what the original intent with that was but it did not seem to provide more functionality, please correct me if Im wrong.

So I finally got all the beamline attributes to work here (on ID29 and mockup ...), set, get and change events. It seems that we need to work a little bit more on abort,  it is not as immediate as one would like (at least in my opinion). A small detail that we need to look at is how to couple the resolution and detector distance in a convenient way.

Thanks,
Marcus

### CODE REVIEW

**Comment by meguiraun** [comment]
hej @marcus-oscarsson, thanks for this, I am looking at this and one thing I notice is that `ui/components/Tasks/DataCollection.js` (and rest of the tasks) need to be adapted.

I'll keep looking and testing

File: mxcube3/ho_mediators/beamline_setup.py [hunk 1]
```diff
@@ -728,7 +728,7 @@ def dict_repr(self):
 class DetectorDistanceHOMediator(HOMediatorBase):
     def __init__(self, ho, name=''):
         super(DetectorDistanceHOMediator, self).__init__(ho, name)
-        ho.connect("positionChanged", self.value_change)
+        #ho.connect("positionChanged", self.value_change)
```
Comment (meguiraun) [inline mxcube3/ho_mediators/beamline_setup.py]: then, how do we notify the client about changes in dtox?

Comment (marcus-oscarsson) [inline mxcube3/ho_mediators/beamline_setup.py]: Well at the moment we don't, its related to what I wrote at the end of my message. We need to find a good way to aggregate the resolution and the detector distance. I added the comment since it was causing me some problems when I was testing, its possible that you are not experiencing the same problem. But we will of course have a way to get those updates

Comment (marcus-oscarsson) [inline mxcube3/ho_mediators/beamline_setup.py]: Ill make a proposal for the resolution and detector distance, I think that will be in another PR if you dont mind.

Comment (meguiraun) [inline mxcube3/ho_mediators/beamline_setup.py]: I have not seen anything on our side, probably because for us dtox is a SardanaMotor. I totally agree on having separate PRs, better that way for everyone ;)

File: mxcube3/ho_mediators/beamline_setup.py [hunk 2]
```diff
@@ -243,6 +269,7 @@ def value_change(self, *args):
         Signal handler to be used for sending values to the client via socketIO,
         data should normally be sent in the "hwr" namespace.
         """
+        print("Emitting beamline value change")
```
Comment (meguiraun) [inline mxcube3/ho_mediators/beamline_setup.py]: I missed this one... could you remove the print? (sorry for being picky)

File: mxcube3/ho_mediators/beamline_setup.py [hunk 3]
```diff
@@ -383,10 +413,8 @@ def limits(self):
         :returns: The limits.
         """
         try:
-            energy_limits = self._ho.getLimits()
+            energy_limits = self._ho.getEnergyLimits()
```
Comment (meguiraun) [inline mxcube3/ho_mediators/beamline_setup.py]: don't we need to do the unit conversion?

Comment (marcus-oscarsson) [inline mxcube3/ho_mediators/beamline_setup.py]: We dont have a method called getLimits, simply getEnergyLimits

Comment (marcus-oscarsson) [inline mxcube3/ho_mediators/beamline_setup.py]: You seem to suggest that you do some additional unit conversion ?

**Comment by marcus-oscarsson** [comment]
Okey so made the necessary changes to "aggregate" the resolution and detector distance. They are actually already aggregated by the resolution object so its just a matter of retrieving the detector distance through the resolution. So Im doing that now, which is at least better than before. Sorry for not waiting with that for another PR as we said, it was such a small change so it was simply better to do it right away.

File: mxcube3/ho_mediators/beamline_setup.py (hunk 3 above)
Comment (meguiraun) [inline mxcube3/ho_mediators/beamline_setup.py]: I would say return the wavelength limits in A instead of eV. How is it done now?

**Comment by meguiraun** [comment]
Thanks @marcus-oscarsson, just one more thing... `ui/components/Tasks/validate.js` needs also fixes. A refactoring is never easy...

No problem with the resolution/detdist, I though you were thinking in a bigger change.

File: mxcube3/ho_mediators/beamline_setup.py (hunk 3 above)
Comment (marcus-oscarsson) [inline mxcube3/ho_mediators/beamline_setup.py]: Ahh, yes for the wavelength object sorry missed that, yes its getWavelengthLimits
